File: Programmers/Lv.0/수열과 구간 쿼리 2/181923.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        
        for (int idx=0; idx<queries.length; idx++) {
            int s = queries[idx][0];
            int e = queries[idx][1];
            int k = queries[idx][2];
            
            int num = -1;
            for (int i=s; i<=e; i++) {
                if (arr[i] > k && (num == -1 || num > arr[i])) {
                    num = arr[i];
                }
            }
            
            answer[idx] = num;
        }
        
        return answer;
    }
}

File: Programmers/Lv.0/수열과 구간 쿼리 2/181923.js
function solution(arr, queries) {
    const answer = [];
    
    for (const [s, e, k] of queries) {
        let num = -1;
        for (let i=s; i<=e; i++) {
            if (arr[i] > k && (num === -1 || num > arr[i])) {
                num = arr[i];
            }
        }
        
        answer.push(num);
    }
    
    return answer;
}
